presto-spi/src/main/java/com/facebook/presto/spi/session/SessionConfigurationContext.java

```diff
@@ -25,12 +27,14 @@
     private final String user;
     private final Optional<String> source;
     private final Set<String> clientTags;
+    private final ResourceGroupId resourceGroupId;
 
-    public SessionConfigurationContext(String user, Optional<String> source, Set<String> clientTags)
+    public SessionConfigurationContext(String user, Optional<String> source, Set<String> clientTags, ResourceGroupId resourceGroupId)
     {
         this.user = requireNonNull(user, "user is null");
         this.source = requireNonNull(source, "source is null");
         this.clientTags = unmodifiableSet(new HashSet<>(requireNonNull(clientTags, "clientTags is null")));
+        this.resourceGroupId = requireNonNull(resourceGroupId, "resourceGroupId");
```

Comment:
Are you sure this works? I recall seeing places in the code where resource group ID is optional. If it is not optional, that might a be a big change. Can you validate this?

Reply:
@raghavsethi : `resourceGroupId` is required to be non null when creating`SelectionContext`:

https://github.com/prestodb/presto/blob/ba4540c969a337a325ca4ac7dc67ea81deda4a28/presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/SelectionContext.java#L23-L27

I think one major reason for using `Optional<ResourceGroupId>` is because `QueryExecution.getResourceGroup` returns an optional. This is because the `resourceGroup` is set after query is run.

Reply:
I think that after your change, we should have resource group ID 100% of the time, given that we throw on did not match selection rule.

Reply:
@raghavsethi : Before that it will still throw, it's just throw when submitting the query.